include/kernel.h

```diff
@@ -2013,6 +2014,127 @@ __syscall int k_futex_wake(struct k_futex *futex, bool wake_all);
 /** @} */
 #endif
 
+/**
+ * @defgroup event_apis Event APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/**
+ * Event Structure
+ * @ingroup event_apis
+ */
+
+struct k_event {
+	_wait_q_t         wait_q;
+	uint32_t          events;
+	struct k_spinlock lock;
+};
+
+#define K_EVENT_ACCUMULATE     0x00
+#define K_EVENT_OVERWRITE      0x01
+#define K_EVENT_DELIVERY_MASK  0x01
+
+#define K_EVENT_WAIT_ALL       0x00
+#define K_EVENT_WAIT_ANY       0x01
+#define K_EVENT_WAIT_MASK      0x01
+
+#define K_EVENT_PRE_CLEAR      0x02
+#define K_EVENT_POST_CLEAR     0x04
```

Comment:
This bit API seems a little clumsy to me.  Basically you've got three **boolean** arguments ("accumulate/overwrite", "wait all/any", "clear pre/post").  Why not pass and store them that way instead?  In fact as I read it only two need to be stored in the thread, so that would in fact save two bytes of the thread object depending on where you pack it.